editor/src/components/canvas/controls/new-canvas-controls.tsx

```diff
@@ -266,22 +307,194 @@ const NewCanvasControlsInner = (props: NewCanvasControlsInnerProps) => {
+    }
+  })
+
+  const getCanvasPoint = React.useCallback(
+    (x: number, y: number): CanvasPoint => {
+      return windowToCanvasCoordinates(scale, canvasOffset, windowPoint({ x, y }))
+        .canvasPositionRounded
+    },
+    [scale, canvasOffset],
+  )
+
+  const onWindowMouseUp = React.useCallback(
+    (e: MouseEvent) => {
+      setSelectionAreaStart(null)
+      setSelectionAreaRectangle(null)
+
+      const isSelectingArea = isSelectModeWithArea(editorMode)
+      if (isSelectingArea) {
+        let actions: EditorAction[] = [switchEditorMode(EditorModes.selectMode())]
+        if (
+          selectionAreaStart != null &&
+          localHighlightedViews.length > 0 &&
+          isValidMouseEventForSelectionArea(e)
+        ) {
+          actions.push(selectComponents(localHighlightedViews, false))
+        }
+        dispatch(actions)
+        setLocalHighlightedViews([])
+      }
+    },
+    [
+      dispatch,
+      setSelectionAreaRectangle,
+      selectionAreaStart,
+      localHighlightedViews,
+      editorMode,
+      setLocalHighlightedViews,
+    ],
+  )
+
+  const onMouseUp = React.useCallback(
+    (e: React.MouseEvent<HTMLDivElement>) => {
+      selectModeHooks.onMouseUp(e)
+    },
+    [selectModeHooks],
+  )
+
+  const onWindowMouseMove = React.useCallback(
+    (e: MouseEvent) => {
+      if (selectionAreaStart != null) {
+        const mousePoint = windowPoint({ x: e.clientX, y: e.clientY })
+        const selectionArea = makeSelectionArea(selectionAreaStart, mousePoint)
+
+        // the rectangle displayed on the canvas
+        const selectionAreaRectangle = getSelectionAreaRenderedRect(
+          selectionArea,
+          ref.current?.getBoundingClientRect() ?? null,
+        )
+        setSelectionAreaRectangle(selectionAreaRectangle)
+
+        // the canvas area for selecting elements
+        const selectionAreaCanvasRect = rectangleFromTLBR(
+          getCanvasPoint(selectionArea.x, selectionArea.y),
+          getCanvasPoint(
+            selectionArea.x + selectionArea.width,
+            selectionArea.y + selectionArea.height,
+          ),
+          true,
+        )
+        const elementsUnderSelectionArea = getAllTargetsUnderAreaAABB(
+          storeRef.current.jsxMetadata,
+          localSelectedViews,
+          hiddenInstances,
+          'no-filter',
+          selectionAreaCanvasRect,
+          pathTrees,
+          allElementProps,
+          false,
+        )
+        const filtered = filterUnderSelectionArea(
+          elementsUnderSelectionArea,
+          storeRef.current.jsxMetadata,
+          selectionAreaCanvasRect,
+        )
+
+        setLocalHighlightedViews(filtered)
+      }
+    },
+    [
+      storeRef,
+      setLocalHighlightedViews,
+      setSelectionAreaRectangle,
+      selectionAreaStart,
+      getCanvasPoint,
+      pathTrees,
+      allElementProps,
+      localSelectedViews,
+      hiddenInstances,
+    ],
+  )
+
+  const onMouseMove = React.useCallback(
```

Comment:
I think this can be written back to how it was: just directly pass in selectModeHooks.onMouseMove to the element